File: frontend/src/app/shared/service/admin/inscription/DictionaryAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {DictionaryDto} from 'src/app/shared/model/inscription/Dictionary.model';
import {DictionaryCriteria} from 'src/app/shared/criteria/inscription/DictionaryCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class DictionaryAdminService extends AbstractService<DictionaryDto, DictionaryCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): DictionaryDto {
        return new DictionaryDto();
    }

    public constrcutCriteria(): DictionaryCriteria {
        return new DictionaryCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/dictionary/';
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/admin/inscription/DictionaryAdminService.java
package ma.zs.alc.service.facade.admin.inscription;

import java.util.List;
import ma.zs.alc.bean.core.inscription.Dictionary;
import ma.zs.alc.dao.criteria.core.inscription.DictionaryCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface DictionaryAdminService extends  IService<Dictionary,DictionaryCriteria>  {

    List<Dictionary> findByEtudiantId(Long id);
    int deleteByEtudiantId(Long id);
    long countByEtudiantId(Long id);




}
